contrib/oauth2/mutt_oauth2.py

```diff
@@ -167,7 +170,7 @@ def writetokenfile():
 if args.authflow:
     authflow = args.authflow
 
-baseparams = {'client_id': registration['client_id']}
+baseparams = {'client_id': args.client_id or registration['client_id']}
```

Comment:
Getting it from registration is pointless.
Unless the user modified the script by hardcoding those values, it doesn't do anything and if they did they wouldn't pull in your changes.

If we want to swtich to a version where the users doesn't monkeypatch the script file prior to running it we can simply remove that case.

Reply:
> Getting it from registration is pointless. Unless the user modified the script by hardcoding those values, it doesn't do anything and if they did they wouldn't pull in your changes.

Understood.  But I tried to only add functionality and not remove any, so I left this there for compatibility reasons, if users prefer to edit their secrets into the script (for whatever reasons).  What do you think?  I'm not going to argue much, though.

> If we want to swtich to a version where the users doesn't monkeypatch the script file prior to running it we can simply remove that case.

Honestly, I don't think it makes much sense to modify the current script into that direction.  I don't want to offend anyone, but script is quite messy and IMHO difficult to modify.  I don't know if I could do any better, but I could give it a try.

Reply:
> > Getting it from registration is pointless. Unless the user modified the script by hardcoding those values, it doesn't do anything and if they did they wouldn't pull in your changes.
> 
> Understood.  But I tried to only add functionality and not remove any, so I left this there for compatibility reasons, if users prefer to edit their secrets into the script (for whatever reasons).  What do you think?  I'm not going to argue much, though.

I believe backwards compatibility is not required here. Backwards compatibility for editing your config into the script was broken recently, in favor of cli args that can go directly into your invokation in your .muttrc

Reply:
> I believe keeping backwards compatibility is not required.

Okay, fine with me.  Fixed!